components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/openidconnect/JWTAccessTokenOIDCClaimsHandler.java

```diff
@@ -0,0 +1,291 @@
+import org.wso2.carbon.identity.application.mgt.ApplicationManagementService;
+import org.wso2.carbon.identity.base.IdentityException;
+import org.wso2.carbon.identity.claim.metadata.mgt.ClaimMetadataHandler;
+import org.wso2.carbon.identity.claim.metadata.mgt.exception.ClaimMetadataException;
+import org.wso2.carbon.identity.oauth.common.exception.InvalidOAuthClientException;
+import org.wso2.carbon.identity.oauth.dao.OAuthAppDO;
+import org.wso2.carbon.identity.oauth2.IdentityOAuth2Exception;
+import org.wso2.carbon.identity.oauth2.authz.OAuthAuthzReqMessageContext;
+import org.wso2.carbon.identity.oauth2.internal.OAuth2ServiceComponentHolder;
+import org.wso2.carbon.identity.oauth2.token.OAuthTokenReqMessageContext;
+import org.wso2.carbon.identity.oauth2.util.OAuth2Util;
+import org.wso2.carbon.identity.openidconnect.internal.OpenIDConnectServiceComponentHolder;
+import org.wso2.carbon.identity.organization.management.service.exception.OrganizationManagementException;
+import org.wso2.carbon.user.api.UserStoreException;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static org.wso2.carbon.identity.oauth.common.OAuthConstants.OIDCClaims.ADDRESS;
+import static org.wso2.carbon.identity.oauth.common.OAuthConstants.OIDCClaims.GROUPS;
+
+/**
+ * A class that provides OIDC claims for JWT access tokens.
+ */
+public class JWTAccessTokenOIDCClaimsHandler implements CustomClaimsCallbackHandler {
+
+    private static final Log log = LogFactory.getLog(JWTAccessTokenOIDCClaimsHandler.class);
+
+    private static final String OAUTH2 = "oauth2";
+    private static final String OIDC_DIALECT = "http://wso2.org/oidc/claim";
+
+    @Override
+    public JWTClaimsSet handleCustomClaims(JWTClaimsSet.Builder builder, OAuthTokenReqMessageContext request)
+            throws IdentityOAuth2Exception {
+
+        String clientId = request.getOauth2AccessTokenReqDTO().getClientId();
+        String spTenantDomain = getServiceProviderTenantDomain(request);
+        AuthenticatedUser authenticatedUser = request.getAuthorizedUser();
+
+        Map<String, Object> claims = getJWTAccessTokenUserClaims(authenticatedUser, clientId, spTenantDomain);
+        if (claims == null || claims.isEmpty()) {
+            return builder.build();
+        }
+        handleClaimsFormat(claims, spTenantDomain);
+        return setClaimsToJwtClaimSet(builder, claims);
+    }
+
+    @Override
+    public JWTClaimsSet handleCustomClaims(JWTClaimsSet.Builder builder, OAuthAuthzReqMessageContext request)
+            throws IdentityOAuth2Exception {
+
+        // TODO : Handle claims for consent
+        return builder.build();
+    }
+
+    private Map<String, Object> getJWTAccessTokenUserClaims(AuthenticatedUser authenticatedUser, String clientId,
```

Comment:
Do we need to return as Map<String, Object>, check whether Map<String, String> is possible to return?

Reply:
setClaimsToJwtClaimSet method required to pass Map<String, Object> for the operation